components/feature/webnotifications/src/main/java/mozilla/components/feature/webnotifications/WebNotificationFeature.kt

```diff
@@ -0,0 +1,103 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.webnotifications
+
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.content.Context
+import android.os.Build
+import androidx.core.app.NotificationCompat
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import mozilla.components.browser.icons.BrowserIcons
+import mozilla.components.concept.engine.Engine
+import mozilla.components.concept.engine.webnotifications.WebNotification
+import mozilla.components.concept.engine.webnotifications.WebNotificationDelegate
+import mozilla.components.support.base.log.logger.Logger
+import java.lang.UnsupportedOperationException
+
+private const val NOTIFICATION_CHANNEL_ID = "mozac.feature.webnotifications.generic"
```

Comment:
Other browsers create a channel per website and put them under a notification group. Should we follow the same model?

Reply:
I agree.  We'll need to do this when the push notification feature is implemented.  Currently GV doesn't give us which web site is notifying.  Once we have that information we can create notification groups with channels for each website.  This will be a design decision of push notification.

Reply:
I that case, I think we should probably set up a notification group for now with this channel as the only entry.

Reply:
Sure, I'll create a channel

Reply:
> Currently GV doesn't give us which web site is notifying.

Maybe this is something we should ask GV if they can provide. We can then group these site permissions together along with web push notifications or at least be able to categories them per website in site settings.

I would argue that a user doesn't know or care about the difference between a WebPush and WebNotification. We should delegate blocking per-site.

We can also punt on this for now and investigate it more with https://github.com/mozilla-mobile/android-components/issues/4628.